lib/grp.gi

```diff
@@ -339,6 +339,10 @@ InstallImmediateMethod( IsPerfectGroup,
     0,
     grp -> Size( grp ) = 1 );
 
+InstallMethod( IsPerfectGroup, "for groups having abelian invariants",
+    [ IsGroup and HasAbelianInvariants ],
+    grp -> Length( AbelianInvariants( grp ) ) = 0 );
+
```

Comment:
@fingolfin Ok, so we need a group that is not created as a subgroup of an fp-group, does not know if it is finite but does know its abelian invariants. Perhaps some PSL group constructed by generators over some infinite field? I am not sure how to do that in GAP.